internal/plugintest/working_dir.go

```diff
@@ -135,17 +153,13 @@ func (wd *WorkingDir) ClearPlan() error {
 // Init runs "terraform init" for the given working directory, forcing Terraform
 // to use the current version of the plugin under test.
 func (wd *WorkingDir) Init() error {
-	if _, err := os.Stat(wd.configFilename()); err != nil {
+	if wd.configFilename == "" {
 		return fmt.Errorf("must call SetConfig before Init")
 	}
 
 	return wd.tf.Init(context.Background(), tfexec.Reattach(wd.reattachInfo))
 }
 
-func (wd *WorkingDir) configFilename() string {
-	return filepath.Join(wd.baseDir, ConfigFileName)
-}
```

Comment:
For consistency, I think we should still keep this method around to automatically do the file joining:

```go
func (wd *WorkingDir) configFilename() string {
	return filepath.Join(wd.baseDir, wd.configFilename)
}
```

And set without baseDir via:

```go
if json.Valid(bCfg) {
	wd.configFilename = ConfigFileNameJSON
} else {
	wd.configFilename = ConfigFileName
}
```

Reply:
Unlike WorkingDir.planFilename() -- which is "static" in the sense that it only depends on the Workdir.baseDir, configFilename is now "dynamic". It can be empty (SetConfig() not called yet) or one of the two cases (HCL/JSON).

When I tried your suggestion, configFilename() would:
1. either have to return (string, error) (error in case no filename has been set up yet), or the caller would be responsible,
2. or the caller would be responsible to check the filename field for emptiness.

The first option is of course the clean and correct one, but it felt cumbersome at the only place where this would be called (Init(), see below).

If you feel strongly, I can refactor it to the method that returns (string, error).